src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/BuiltInTriggerTests.cs

```diff
@@ -0,0 +1,172 @@
+                GreaterThan = GCHeapSizeOptionsDefaults.GreaterThan,
+                LessThan = null,
+                SlidingWindowDuration = SlidingWindowDurationDefault
+            };
+
+            await IEventCounterTrigger_CorrectOptions<GCHeapSizeOptions>(KnownCollectionRuleTriggers.GCHeapSize, expectedSettings);
+        }
+
+        [Fact]
+        public async Task ThreadpoolQueueLengthTrigger_CorrectDefaultOptions()
+        {
+            EventCounterOptions expectedSettings = new()
+            {
+                ProviderName = IEventCounterShortcutsConstants.SystemRuntime,
+                CounterName = IEventCounterShortcutsConstants.ThreadpoolQueueLength,
+                GreaterThan = ThreadpoolQueueLengthOptionsDefaults.GreaterThan,
+                LessThan = null,
+                SlidingWindowDuration = SlidingWindowDurationDefault
+            };
+
+            await IEventCounterTrigger_CorrectOptions<ThreadpoolQueueLengthOptions>(KnownCollectionRuleTriggers.ThreadpoolQueueLength, expectedSettings);
+        }
+
+        [Fact]
+        public async Task CPUUsageTrigger_CorrectDefaultOptions()
+        {
+            EventCounterOptions expectedSettings = new()
+            {
+                ProviderName = IEventCounterShortcutsConstants.SystemRuntime,
+                CounterName = IEventCounterShortcutsConstants.CPUUsage,
+                GreaterThan = CPUUsageOptionsDefaults.GreaterThan,
+                LessThan = null,
+                SlidingWindowDuration = SlidingWindowDurationDefault
+            };
+
+            await IEventCounterTrigger_CorrectOptions<CPUUsageOptions>(KnownCollectionRuleTriggers.CPUUsage, expectedSettings);
+        }
+
+        private async Task IEventCounterTrigger_CorrectOptions<T>(string triggerName, EventCounterOptions expectedSettings, Action<IEventCounterShortcuts> callback = null)
+        {
+            using TemporaryDirectory tempDirectory = new(_outputHelper);
+
+            await TestHostHelper.CreateCollectionRulesHost(_outputHelper, rootOptions =>
+            {
+                rootOptions.AddFileSystemEgress(ActionTestsConstants.ExpectedEgressProvider, tempDirectory.FullName);
+
+                rootOptions.CreateCollectionRule(DefaultRuleName)
+                    .AddCollectDumpAction(ActionTestsConstants.ExpectedEgressProvider)
+                    .SetIEventCounterTrigger(typeof(T), triggerName, callback);
+            }, host =>
+            {
+                T options = TriggerTestsHelper.GetTriggerOptions<T>(host, DefaultRuleName);
+
+                ICollectionRuleTriggerFactoryProxy factory;
+                Assert.True(host.Services.GetService<ICollectionRuleTriggerOperations>().TryCreateFactory(triggerName, out factory));
+
+                EventPipeTrigger<EventCounterTriggerSettings> trigger = (EventPipeTrigger<EventCounterTriggerSettings>)factory.Create(new EndpointInfo(), null, options);
```

Comment:
This is the reason for the build failures - I forgot that I had needed to change the visibility of some things in the diagnostics repo to do this, which meant the tests build and succeed locally. Since I'm not sure if those are changes we should be making in that repo, I'll hold off on pushing anything there until I get feedback on this - or maybe there's just a better way to test this.

Reply:
I don't think you should be testing that the settings are correctly set at the pipeline settings level. You should be able to allow the configuration system to process the configuration, pull out the options instance (using something like `IOptions<EventCounterOptions>.Value`), and check its values. Testing the translation from `dotnet monitor` options to pipeline options should not be the responsibility of this test.

Reply:
I think I'm a bit confused on what you're suggesting - is the implication that we could call `optionsMonitor.Get(DefaultRuleName).Trigger.Settings` and check the `EventCounterOptions`? My understanding (and what I've been seeing) is that this would return the original `CPUUsageOptions`, because the translation is only used for the pipeline and not changing the options that we pass into the factory. Hence the current strategy of inspecting the pipeline itself. I realized we could also do testing like `CollectionRulePipeline_EventCounterTriggerTest` but for the built-in shortcuts.

Reply:
If you are getting the rule options through an IOptions*<CollectionRuleOptions> instance, that means that `IConfigureOptions<CollectionRuleOptions>`, `IPostConfigureOptions<CollectionRuleOptions>`, and `IValidateOptions<CollectionRuleOptions>` have already run. I would expect that the `CollectionRuleOptions` instance you get should be completely filled at that point and not require further mutation of anything on it. That means the `CPUUsageOptions` should have already been translated to `EventCounterOptions` and placed into the `ruleOptions.Trigger.Settings` property. You should be able to use that property path to validate that the options were set correctly from the shortcut options.

Reply:
Ah - in which case, I think the change needs to be made in the implementation, and not just the testing. Instead of doing the translation in `EventCounterTriggerFactory` on create, moving the translation logic into `IConfigureOptions` or `IPostConfigureOptions` would actually change the `CollectionRuleOptions`, in which case `EventCounterTriggerFactory` doesn't need to handle any of the different `IEventCounterShortcuts`, assuming that's the desired behavior.

Reply:
That's what I would expect. I haven't looked at implementation, only the test code you pointed out.